Khulnsoft Networking Patch
=======================

This patch modifies the standard libraries roundTrip function on the HTTP interface, such that Khulnsoft's
`khulnsoftBeginRoundTrip` and `khulnsoftFinishRoundTrip` are called. These calls are routed into Khulnsoft runtime and allows us
instrument the call.

diff --git a/src/net/http/transport.go b/src/net/http/transport.go
index da9163a27a..b2200d5952 100644
--- a/src/net/http/transport.go
+++ b/src/net/http/transport.go
@@ -527,9 +527,17 @@ func validateHeaders(hdrs Header) string {
 }

 // roundTrip implements a RoundTripper over HTTP.
-func (t *Transport) roundTrip(req *Request) (_ *Response, err error) {
+func (t *Transport) roundTrip(req *Request) (rr *Response, err error) {
 	t.nextProtoOnce.Do(t.onceSetNextProtoDefaults)
-	ctx := req.Context()
+	ctx, err := khulnsoftBeginRoundTrip(req)
+	if err != nil {
+		req.closeBody()
+		return nil, err
+	}
+	req.ctx = ctx
+	defer func() {
+		khulnsoftFinishRoundTrip(req, rr, err)
+	}()
 	trace := httptrace.ContextClientTrace(ctx)

 	if req.URL == nil {
